app/models/otp_requests_tracker.rb

```diff
@@ -0,0 +1,21 @@
+class OtpRequestsTracker < ActiveRecord::Base
+  include EncryptableAttribute
+
+  encrypted_attribute_without_setter(name: :phone)
+
+  def self.find_or_create_with_phone(phone)
+    tries ||= 1
+    phone ||= phone.strip
+    phone_fingerprint ||= Pii::Fingerprinter.fingerprint(phone)
+
+    where(phone_fingerprint: phone_fingerprint).
+      first_or_create(phone: phone, otp_send_count: 0, otp_last_sent_at: Time.zone.now)
+  rescue ActiveRecord::RecordNotUnique
+    retry unless (tries -= 1).zero?
```

Comment:
We need to `raise` when we're not retrying otherwise this will just return `nil` and we'll get a `NoMethodError` rather than a duplicate key error

so like
```
rescue ActiveRecord::RecordNotUnique
  retry unless ...
  raise
end
```